astropy/io/fits/tests/test_hdulist.py

```diff
@@ -913,3 +914,38 @@ def test_proper_error_raised_on_non_fits_file_with_unicode(self):
         # This should raise an IOError because there is no end card.
         with pytest.raises(IOError):
             fits.open(filename)
+
+    # Avoid a spurious pyflakes warning about an unused import
+    # of six by adding a reason string
+    @pytest.mark.skipif(six.PY2, reason='six.PY2')
```

Comment:
To avoid `pyflakes` warning, you could have just write `from ....extern import six  # noqa` above. The reason here should be used to explain why the test is skipped for Python 2 (although I know you explained it in the docstring).

But I am okay with leaving this as-is because Python 2 support will be removed in about 3 months anyway.